src/tests/ftest/soak/soak.py

```diff
@@ -65,120 +72,228 @@ def job_done(self, args):
-                    cmd = "srun -l --mpi=pmi2 --export={} {}".format(
-                        ",".join(exports), ior_cmd)
-                    command.append(cmd)
-                    self.log.debug("<<IOR cmdline >>: %s \n", cmd)
-        return command
-
-    def create_dmg_cmdline(self, job_params, job_spec, pool):
+
+        api_list = self.params.get("api", ior_params + "*")
+        tsize_list = self.params.get("transfer_size", ior_params + "*")
+        bsize_list = self.params.get("block_size", ior_params + "*")
+        oclass_list = self.params.get("daos_oclass", ior_params + "*")
+        # check if capable of doing rebuild; if yes then daos_oclass = RP_*GX
+        if self.is_harasser("rebuild"):
+            oclass_list = self.params.get("daos_oclass", "/run/rebuild/*")
+        # update IOR cmdline for each additional IOR obj
+        for task in self.task_list:
+            for api in api_list:
+                for b_size in bsize_list:
+                    for t_size in tsize_list:
+                        for o_type in oclass_list:
+                            ior_cmd = IorCommand()
+                            ior_cmd.namespace = ior_params
+                            ior_cmd.get_params(self)
+                            if iteration is not None and iteration < 0:
+                                ior_cmd.repetitions.update(1000000)
+                            if self.job_timeout is not None:
+                                ior_cmd.max_duration.update(self.job_timeout)
+                            else:
+                                ior_cmd.max_duration.update(10)
+                            ior_cmd.api.update(api)
+                            ior_cmd.block_size.update(b_size)
+                            ior_cmd.transfer_size.update(t_size)
+                            ior_cmd.daos_oclass.update(o_type)
+                            ior_cmd.set_daos_params(self.server_group, pool)
+
+                            # slurm/sbatch cmdline
+                            if self.job_manager == "Sbatch":
+                                env = ior_cmd.get_default_env("srun", self.tmp)
+                                if ior_cmd.api.value == "MPIIO":
+                                    env["DAOS_CONT"] = ior_cmd.daos_cont.value
+                                cmd = Srun(ior_cmd)
+                                cmd.setup_command(env, None, None)
+                                commands.append(
+                                    [job_spec, cmd.__str__(), task])
+                            else:
+                                raise SoakTestError(
+                                    "<<FAILED: {} is not supported.".format(
+                                        self.job_manager))
+                            self.log.info("<<IOR cmdline>>: %s \n",
+                                          commands[-1][1].__str__())
+        return commands
+
+    def create_dmg_cmdline(self, job_spec, pool):
```

Comment:
(pylint-unused-argument) Unused argument 'job_spec'
(pylint-unused-argument) Unused argument 'pool'
(pylint-no-self-use) Method could be a function